tests/project_template/test_settings.py

```diff
@@ -1,30 +1,33 @@
 import os
 import shutil
+import tempfile
 
 from django import conf
 from django.test import TestCase
+from django.test.utils import extend_sys_path
 
 
 class TestStartProjectSettings(TestCase):
     def setUp(self):
-        # Ensure settings.py exists
-        project_dir = os.path.join(
+        self.temp_dir = tempfile.TemporaryDirectory()
+        self.addCleanup(self.temp_dir.cleanup)
```

Comment:
It looks like relying on the finalizer would result in a warning:

https://github.com/python/cpython/blob/1e62bf145b4865d03a29a5720a4eb84c321a9829/Lib/tempfile.py#L791-L798

Further, it is my understanding that the Python stdlib is designed such that all resources should be explicitly cleaned up. They typically result in a warning if you do not.

The [Python docs](https://docs.python.org/3/reference/datamodel.html#object.__del__) also contain this note:

> It is not guaranteed that `__del__()` methods are called for objects that still exist when the interpreter exits.